src/server/scripts/Northrend/AzjolNerub/Ahnkahet/boss_amanitar.cpp

```diff
@@ -19,218 +19,298 @@
+                    while (!_mushroomsQueue.empty())
+                    {
+                        SpawnMushroom(_mushroomsQueue.front());
+                        _mushroomsQueue.pop();
+                    }
+                    events.Repeat(Seconds(40), Seconds(60));
+                    break;
+                default:
+                    break;
             }
-        };
 
-        CreatureAI* GetAI(Creature* creature) const override
-        {
-            return GetAhnKahetAI<boss_amanitarAI>(creature);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
         }
+
+        DoMeleeAttackIfReady();
+    }
+
+    private:
+        std::queue<uint32> _mushroomsQueue;
 };
 
-class npc_amanitar_mushrooms : public CreatureScript
+struct npc_amanitar_mushrooms : public ScriptedAI
 {
-public:
-    npc_amanitar_mushrooms() : CreatureScript("npc_amanitar_mushrooms") { }
+    npc_amanitar_mushrooms(Creature* creature) : ScriptedAI(creature), _active(false), _mySpot(0) { }
 
-    struct npc_amanitar_mushroomsAI : public ScriptedAI
+    void Reset() override
     {
-        npc_amanitar_mushroomsAI(Creature* creature) : ScriptedAI(creature) { }
+        me->SetReactState(REACT_PASSIVE);
+        me->SetDisplayId(me->GetCreatureTemplate()->Modelid2);
+        DoCastSelf(SPELL_PUTRID_MUSHROOM);
+        DoCastSelf(SPELL_SHRINK, true);
+        DoCastSelf(SPELL_GROW, true);
 
-        EventMap events;
+        if (me->GetEntry() == NPC_HEALTHY_MUSHROOM)
+        {
+            DoCastSelf(SPELL_POWER_MUSHROOM_VISUAL_AURA);
+            _active = true;
+        }
+        else
+            DoCastSelf(SPELL_POISONOUS_MUSHROOM_VISUAL_AURA);
 
-        void Reset() override
+        _scheduler.Schedule(Milliseconds(800), [this](TaskContext /*context*/)
         {
-            events.Reset();
-            events.ScheduleEvent(EVENT_AURA, 1 * IN_MILLISECONDS);
+            DoCastSelf(SPELL_GROW, true);
+        });
+    }
+
+    void MoveInLineOfSight(Unit* target) override
+    {
+        if (_active || target->GetTypeId() != TYPEID_PLAYER || me->GetDistance2d(target) > 2.0f)
+            return;
 
-            me->SetDisplayId(me->GetCreatureTemplate()->Modelid2);
-            DoCast(SPELL_PUTRID_MUSHROOM);
+        _active = true;
 
-            if (me->GetEntry() == NPC_POISONOUS_MUSHROOM)
-                DoCast(SPELL_POISONOUS_MUSHROOM_VISUAL_AURA);
-            else
-                DoCast(SPELL_POWER_MUSHROOM_VISUAL_AURA);
-        }
+        target->RemoveAurasDueToSpell(SPELL_POTENT_FUNGUS);
+        DoCastAOE(SPELL_POISONOUS_MUSHROOM_POISON_CLOUD);
 
-        void DamageTaken(Unit* /*attacker*/, uint32 &damage) override
+        _scheduler.Schedule(Seconds(1), [this](TaskContext /*context*/)
         {
-            if (damage >= me->GetHealth() && me->GetEntry() == NPC_HEALTHY_MUSHROOM)
-                DoCast(me, SPELL_HEALTHY_MUSHROOM_POTENT_FUNGUS, true);
-        }
+            me->SetObjectScale(0.1f);
```

Comment:
no spell involved?

Reply:
nothing shown on sniffs

Reply:
I hope it still works to kill the Healthy Mushroom to gain the Potent Fungus to remove the Mini debuff (https://wow.gamepedia.com/Amanitar#Abilities).